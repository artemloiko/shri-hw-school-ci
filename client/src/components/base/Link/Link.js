import React from 'react';
import PropTypes from 'prop-types';
import { Link as RouterLink } from '@reach/router';
import { cn } from 'utils/bem-cn';

import './Link.css';

function Link(props) {
  const { to = '', children } = props;

  return (
    <RouterLink to={to} className={cn('link', props)}>
      {children}
    </RouterLink>
  );
}

Link.propTypes = {
  to: PropTypes.string,
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  mix: PropTypes.arrayOf(PropTypes.string),
};

export default Link;
