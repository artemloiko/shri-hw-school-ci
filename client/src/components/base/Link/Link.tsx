import React from 'react';
import { Link as RouterLink } from '@reach/router';
import { cn, CNProps } from '../../../utils/bem-cn';

import './Link.css';

export type LinkProps = { to?: string; children: any } & CNProps;

const Link: React.FC<LinkProps> = (props) => {
  const { to = '', children } = props;

  return (
    <RouterLink to={to} className={cn('link', props)}>
      {children}
    </RouterLink>
  );
};

export default Link;
