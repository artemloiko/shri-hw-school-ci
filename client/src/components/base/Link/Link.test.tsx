import React from 'react';
import renderer from 'react-test-renderer';

import Link from './Link';

test('Link with default props matches the snapshot', () => {
  const tree = renderer.create(<Link to="/path">Link</Link>).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Link with all possible props matches the snapshot', () => {
  const tree = renderer
    .create(
      <Link className="block" mix={['block']} to="/path">
        Link
      </Link>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
