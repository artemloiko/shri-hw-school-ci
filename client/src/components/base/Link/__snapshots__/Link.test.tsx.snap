// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link with all possible props matches the snapshot 1`] = `
<a
  className="block link block__link"
  href="/path"
  onClick={[Function]}
>
  Link
</a>
`;

exports[`Link with default props matches the snapshot 1`] = `
<a
  aria-current="page"
  className="link"
  href="/"
  onClick={[Function]}
>
  Link
</a>
`;
